### ISSUE TITLE
get_window_size returns bogus values

### ISSUE THREAD

**Issue by agersant** [Gekkio/imgui-rs#175]
Hi there!

First of all, thank you for writing and maintaining imgui-rs, the API is a pleasure to use and it works pretty great.

I am seeing a strange issue where get_window_size() seems to return garbage/uninitialized values whenever I call it. I tried both within a Window's build closure and within a Main Menu closure. Below are the corresponding code snippets I am using. I believe it is a reasonable pattern of usage based on https://github.com/ocornut/imgui/issues/252.

The print statements show seemingly random values, usually very near 0 but sometimes gigantic (or even negative). I am using the latest (as of today) version of imgui-rs from master.

```
ui.with_style_vars(&vec![WindowRounding(0.0), WindowBorderSize(0.0)], || {
	ui.window(im_str!("Documents"))
		.position(rect.position, ImGuiCond::FirstUseEver)
		.always_auto_resize(true)
		.collapsible(false)
		.resizable(false)
		.title_bar(false)
		.menu_bar(false)
		.movable(false)
		.build(|| {
			for document in state.documents_iter() {
				if ui.small_button(&ImString::new(document.get_source().to_string_lossy())) {
					commands.focus_document(document);
				}
				ui.same_line(0.0);
			}
			println!("DOCUMENTS SIZE {:?}", ui.get_window_size());
		});
});
```

```
ui.main_menu_bar(|| {
	ui.menu(im_str!("File")).build(|| {
		if ui.menu_item(im_str!("New Sheet…")).build() {
			commands.new_document();
		}
		if ui.menu_item(im_str!("Open Sheet…")).build() {
			commands.open_document();
		}
		ui.separator();
		if ui.menu_item(im_str!("Save")).build() {
			commands.save();
		}
		if ui.menu_item(im_str!("Save As…")).build() {
			commands.save_as();
		}
		if ui.menu_item(im_str!("Save All")).build() {
			commands.save_all();
		}
		ui.separator();
		if ui.menu_item(im_str!("Close")).build() {
			commands.close_current_document();
		}
		if ui.menu_item(im_str!("Close All")).build() {
			commands.close_all_documents();
		}
	});
	ui.menu(im_str!("View")).build(|| {
		ui.menu_item(im_str!("Grid")).build();
		ui.menu_item(im_str!("Hitboxes")).build();
	});
	ui.menu(im_str!("Help")).build(|| {
		ui.menu_item(im_str!("About")).build();
	});

	println!("MAIN MENU SIZE {:?}", ui.get_window_size());
});
```

**Comment by agersant** [comment]
This is using latest from master (77c79f4e612a8183f8ed6bd78fc68fbc0afaee71), with the gfx backend.

I just noticed the same thing is happening for calls to `get_content_region_avail`.

**Comment by Gekkio** [comment]
Are you a Mac OS user by any chance? It's clearly some memory layout / call convention issue, and I've got a hunch about the problem but can't reproduce the issue on my machine.

**Comment by agersant** [comment]
No, this is on Windows 10 using the msvc ABI toolchain.

I would be happy to help narrowing down the issue but I'm not sure how to.

**Comment by Gekkio** [comment]
Ok, MSVC can also have some surprises :smile: 

Could you try the [tmp/test_175](https://github.com/Gekkio/imgui-rs/tree/tmp/test_175) branch and check if `get_window_pos` and/or `get_window_size` work as they should?

**Comment by Gekkio** [comment]
Ok, looks like it's a C ABI compatibility problem involving C++ UDTs (User Defined Types). I'll try to get this fixed next week

**Comment by Gekkio** [comment]
Ok, this problem should now be fixed in master by eea2c8b00e485c8d41a2ab6c005c9dd81474bcd9

**Comment by agersant** [comment]
Everything working as expected, thank you very much!
